processor/src/main/java/org/mapstruct/ap/internal/model/source/MapMapping.java

```diff
@@ -78,36 +80,45 @@ public static MapMapping fromPrism(MapMappingPrism mapMapping, ExecutableElement
             mapMapping.valueQualifiedByName(),
             valueTargetTypeIsDefined ? mapMapping.valueTargetType() : null);
 
-        return new MapMapping(
+        FormattingParameters keyFormatting = new FormattingParameters(
             mapMapping.keyDateFormat(),
-            keySelection,
+            mapMapping.keyNumberFormat() );
```

Comment:
No. I reckon its (silently) ignored. Perhaps such checking should be in relation to the source type and not in relation to each-other.